bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/magento/GraphqlDataServiceImpl.java

```diff
@@ -159,13 +157,14 @@ public ProductInterface getProductBySku(String sku, String storeView) {
         }
 
         try {
-            return productCache.get(sku, () -> getProductBySkuImpl(sku, storeView)).orElse(null);
+            String key = sku + StringUtils.defaultString(storeView, MAGENTO_DEFAULT_STORE);
```

Comment:
It would be better to use a separator when generating the key. Otherwise duplicate keys because of overlaps in the sku and storeview might occur.

Reply:
I generated all cache keys with the same method so it's kind of unified and now used a separator that is less likely (than `-`) to appear in skus and store views but ... this doesn't prevent collisions, it just makes them less likely to occur. Consider the following cases:

```
   SKU        STORE      SEPARATOR     CACHE KEY

sku|        store|view      ''       sku|store|view
sku|store   |view           ''       sku|store|view
sku         store|view      '|'      sku|store|view
sku|store   view            '|'      sku|store|view
```
They all generate the same keys, so using a separator doesn't really help. However I think the `|` character wouldn't be common in skus and store views, so I think we can consider the collision probability very low.

Reply:
@herzog31 I finally decided to replace the string keys with a dedicated class to use an array as a key, so we completely remove the issue with strings.